esphome/components/esp32/__init__.py

```diff
@@ -169,9 +169,28 @@ def add_idf_component(
             KEY_REF: ref,
             KEY_PATH: path,
             KEY_REFRESH: refresh,
-            KEY_COMPONENTS: components,
-            KEY_SUBMODULES: submodules,
+            KEY_COMPONENTS: components if components is not None else [],
```

Comment:
Line 164-165 above ensures it is at least an empty list already
```suggestion
            KEY_COMPONENTS: components,
```